tests/data_context/test_data_context_utils.py

```diff
@@ -286,6 +287,222 @@ def test_password_masker_mask_db_url(monkeypatch, tmp_path):
+            if (
+                conn_string_password
+                in unaltered_datasources[name]["execution_engine"]["connection_string"]
+            ):
+                # it does contain a password, so make sure its masked
+                assert (
+                    conn_string_password
+                    not in processed_config["execution_engine"]["connection_string"]
+                )
+            else:
+                # it doesn't contain a password, so make sure it's unaltered
+                assert processed_config == unaltered_datasources[name]
+
+        # processed_config either doesn't have an `execution_engine` field,
+        # or a `connection_string` field
+        else:
+            # expect this config to be unaltered
+            assert processed_config == unaltered_datasources[name]
+
+
+def test_sanitize_config_with_no_sensitive_keys():
+    """Test case for PasswordPasker.sanitize_config"""
+
+    # base case - this config should pass through unaffected
+    config = {
+        "some_field": "and a value",
+        "some_other_field": {"password": "but this won't be found"},
+    }
+    config_copy = safe_deep_copy(config)
+    assert PasswordMasker.sanitize_config(config_copy) == config
+
+
+def test_sanitize_config_with_password_field():
+    """Test case for PasswordPasker.sanitize_config"""
+
+    # this case has a password field inside a credentials dict - expect it to be masked
+    config = {"credentials": {"password": "my-super-duper-secure-passphrase-123"}}
+    config_copy = safe_deep_copy(config)
+    res = PasswordMasker.sanitize_config(config_copy)
+    assert res != config
+    assert res["credentials"]["password"] == PasswordMasker.MASKED_PASSWORD_STRING
+
+
+def test_sanitize_config_with_url_field(
+    conn_string_with_embedded_password, conn_string_password
+):
+    """Test case for PasswordPasker.sanitize_config"""
+
+    # this case has a url field inside a credentials dict - expect the password inside
+    # of it to be masked
+    config = {"credentials": {"url": conn_string_with_embedded_password}}
+    config_copy = safe_deep_copy(config)
+    res = PasswordMasker.sanitize_config(config_copy)
+    assert res != config
+    assert conn_string_password not in res["credentials"]["url"]
+    assert PasswordMasker.MASKED_PASSWORD_STRING in res["credentials"]["url"]
+
+
+def test_sanitize_config_with_non_sensitive_url_field():
+    """Test case for PasswordPasker.sanitize_config"""
+
+    # this case has a BigQuery url field inside a credentials dict, which doesn't have a
+    # password - expect it to be untouched
+    config = {"credentials": {"url": "bigquery://foo/bar"}}
+    config_copy = safe_deep_copy(config)
+    res = PasswordMasker.sanitize_config(config_copy)
+    assert res == config
+
+
+def test_sanitize_config_with_nested_url_field(
+    conn_string_password, conn_string_with_embedded_password
+):
+    """Test case for PasswordPasker.sanitize_config"""
```

Comment:
Typo in all docstrings.

Nitpick - since the name of the test specifies the method in question, is this necessary? I think you could probably get by without it since these are very simple (and well written!) unit test.